Core/Rewriters/ExpectCallRewriter.cs

```diff
@@ -0,0 +1,153 @@
+      }
+      catch (ArgumentException ex)
+      {
+        Console.Error.WriteLine ($"{ex}\r\nUnable to rewrite Expect.Call Statement in {node.SyntaxTree.FilePath}");
+        throw;
+      }
+
+      if (node is ExpressionStatementSyntax expressionStatement)
+      {
+        return expressionStatement.WithExpression (
+            (ExpressionSyntax) RewriteExpectCall (expressionStatement.Expression, callSymbols, simpleRhinoMocksSymbols, constraintsSymbols));
+      }
+
+      if (node is not InvocationExpressionSyntax rhinoMocksInvocationExpression
+          || rhinoMocksInvocationExpression.Expression is not MemberAccessExpressionSyntax rhinoMocksMemberAccessExpression)
+      {
+        return node;
+      }
+
+      if (callSymbols.Contains (symbol, SymbolEqualityComparer.Default))
+      {
+        return ConvertExpectExpression (rhinoMocksInvocationExpression);
+      }
+
+      if (constraintsSymbols.Contains (symbol, SymbolEqualityComparer.Default))
+      {
+        var rewrittenContainedExpression = RewriteExpectCall (rhinoMocksMemberAccessExpression.Expression, callSymbols, simpleRhinoMocksSymbols, constraintsSymbols);
+        return ConvertMockedExpression (rewrittenContainedExpression, rhinoMocksMemberAccessExpression, rhinoMocksInvocationExpression);
+      }
+
+      if (simpleRhinoMocksSymbols.Contains (symbol, SymbolEqualityComparer.Default))
+      {
+        return rhinoMocksInvocationExpression.ReplaceNode (
+            rhinoMocksMemberAccessExpression.Expression,
+            (ExpressionSyntax) RewriteExpectCall (rhinoMocksMemberAccessExpression.Expression, callSymbols, simpleRhinoMocksSymbols, constraintsSymbols));
+      }
+
+      throw new InvalidOperationException ("This code should never be executed!");
+    }
+
+    private static IEnumerable<ISymbol> GetAllSimpleRhinoMocksSymbols (INamedTypeSymbol rhinoMocksIMethodOptionsSymbol)
+    {
+      return rhinoMocksIMethodOptionsSymbol.GetMembers ("Return")
+          .Concat (rhinoMocksIMethodOptionsSymbol.GetMembers ("WhenCalled"))
+          .Concat (rhinoMocksIMethodOptionsSymbol.GetMembers ("Callback"))
+          .Concat (rhinoMocksIMethodOptionsSymbol.GetMembers ("IgnoreArguments"));
+    }
+
+    private SyntaxNode ConvertMockedExpression (
+        SyntaxNode rewrittenContainedExpression,
+        MemberAccessExpressionSyntax originalRhinoMocksMemberAccessExpression,
+        InvocationExpressionSyntax originalRhinoMocksInvocationExpression)
+    {
+      var rewrittenInvocationExpression =
+          (InvocationExpressionSyntax) ((SimpleLambdaExpressionSyntax) rewrittenContainedExpression.GetFirstArgument().Expression).ExpressionBody!;
+
+      var oldInvocationExpression = (InvocationExpressionSyntax) originalRhinoMocksMemberAccessExpression.Expression.GetFirstArgument().Expression;
+      var rewrittenArgumentList = ConvertConstraints (originalRhinoMocksInvocationExpression.ArgumentList, GetMethodParameterTypes (oldInvocationExpression).ToList());
```

Comment:
There are 2 words for the same kind of thing: `old`/`original`. You should settle for one in a file (better in the whole solution).